apm-agent-attach-cli/src/main/java/co/elastic/apm/attach/UserRegistry.java

```diff
@@ -222,5 +249,111 @@ public boolean isCurrentUser() {
+            ProcessBuilder buildTheProcess = new ProcessBuilder(command);
+            // merge stdout and stderr so we only have to read one stream
+            buildTheProcess.redirectErrorStream(true);
+            Process spawnedProcess = null;
+            int exitValue = -1;
+            Exception exception = null;
+            StringBuilder commandOutput = new StringBuilder();
+            try {
+                spawnedProcess = buildTheProcess.start();
+
+                long timeout = 5 * 1000L; // NOTE 5 second timeout!
+                long start = System.currentTimeMillis();
+                long now = start;
+                boolean isAlive = true;
+                byte[] buffer = new byte[64 * 1000];
+                try (InputStream in = spawnedProcess.getInputStream()) {
+                    // stop trying if the time elapsed exceeds the timeout
+                    while (isAlive && (now - start) < timeout) {
+                        while (in.available() > 0) {
+                            int lengthRead = in.read(buffer, 0, buffer.length);
+                            commandOutput.append(new String(buffer, 0, lengthRead));
+                        }
+                        try {
+                            Thread.sleep(50);
+                        } catch (InterruptedException e) {
+                            //no action, just means the next loop iteration is earlier
+                        }
+                        now = System.currentTimeMillis();
+                        // if it's not alive but there is still readable input, then continue reading
+                        isAlive = spawnedProcess.isAlive() || in.available() > 0;
+                    }
+                }
+                //would like to call waitFor(TIMEOUT) here, but that is 1.8+
+                //so will pause then have a go at getting the exit value
+                try {
+                    Thread.sleep(50);
+                } catch (InterruptedException e) {
+                    //no action, just means the next loop iteration is earlier
+                }
+                try {
+                    exitValue = spawnedProcess.exitValue();
+                } catch (IllegalThreadStateException e2) {
+                    //ignore and let the finally block have the last say
+                }
+
+                return new CommandOutput(commandOutput, exitValue, exception);
+            } catch (IOException e1) {
+                exception = e1;
+                return new CommandOutput(commandOutput, exitValue, exception);
+            } finally {
+                // Cleanup as well as we can
+                if (spawnedProcess != null && spawnedProcess.isAlive()) {
+                    spawnedProcess.destroy();
+                    try {
+                        Thread.sleep(50);
+                    } catch (InterruptedException e) {
+                        //no action, just means the next loop iteration is earlier
```

Comment:
[minor] which loop?

Reply:
```suggestion
                        //no action, just means the next loop iteration checking
                        //for timeout or process dead, is earlier
```

Reply:
clarified in next commit

Reply:
fixed in commit 3a02a5cd094fd7eb4b033d3167e302463b90bf70